tracer/src/Datadog.Trace/Agent/Transports/HttpClientRequestFactory.cs

```diff
@@ -12,7 +12,11 @@
 
 namespace Datadog.Trace.Agent.Transports
 {
+#if NET5_0_OR_GREATER // in .NET 6 we derive a SocketHandlerRequestFactory
```

Comment:
In .NET 6 we derive from this to create `SocketHandlerRequestFactory` so we _can't_ be sealed